### PR TITLE
Add App Instance status in info msg

### PR DESCRIPTION
**Pull request by kalyan-nidumolu**
Still testing. Will complete the tests shortly

### CODE REVIEW

File: api/proto/info/info.proto [hunk 1]
```diff
@@ -258,6 +264,7 @@ message ZInfoDevice {
   DataSecAtRest dataSecAtRestInfo = 29; //Info about Data At Rest Security
 
   ZInfoConfigItemStatus configItemStatus = 31;
+  repeated ZInfoAppInstance appInstances = 41;
```
Comment (eriknordmark) [inline api/proto/info/info.proto]: Probably better to pack the tags i.e. pick 32.

File: api/proto/info/info.proto [hunk 2]
```diff
@@ -221,6 +221,12 @@ message ZInfoConfigItemStatus {
     map<string, ZInfoConfigItem> unknownConfigItems = 2;
 }
 
+message ZInfoAppInstance {
```
Comment (eriknordmark) [inline api/proto/info/info.proto]: Please add a comment the intended use.

File: api/proto/info/info.proto [hunk 3]
```diff
@@ -221,6 +221,12 @@ message ZInfoConfigItemStatus {
...
+    string uuid = 1;
+    string name = 2;
+    string domainName = 3;
+}
+
 // Base device info, as discovered by Xen (or OS on bare metal)
 message ZInfoDevice {
```
Comment (eriknordmark) [inline api/proto/info/info.proto]: FWIW I was inclined to ask to add state here, but then a realize with that we'd need to trigger a device info message each time the app info state changes. So not adding it means we only need to trigger when the AI is created and deleted which is less traffic.

File: api/proto/info/info.proto (hunk 1 above)
Comment (kalyan-nidumolu) [reply api/proto/info/info.proto]: I would like to put related fields together as much as possible, so leave a bit of gap between unrelated fields for additions .. Hence did this.

File: api/proto/info/info.proto (hunk 2 above)
Comment (kalyan-nidumolu) [reply api/proto/info/info.proto]: Done..

File: api/proto/info/info.proto (hunk 3 above)
Comment (kalyan-nidumolu) [reply api/proto/info/info.proto]: I was also very tempted about it.. But for my current usage - I didn't need it - and hence did not addit in.

But in another way - it is actually good to send these more granular states - debugging would be much easier. The Device / App Instance events generated in the cloud would also be much more accurate.. DownSide is - more data would be sent on low bandwidth links..

Comment (kalyan-nidumolu) [reply api/proto/info/info.proto]: Hmm.. Looking at the code - we are triggering an update again anyways - in Modify.. So might as well add the State.. Will add it..

**Comment by kalyan-nidumolu** [comment]
> You also need a triggerdeveinfo when the app instance is created and deleted.

Good point.. Added it.

File: api/proto/info/info.proto (hunk 3 above)
Comment (eriknordmark) [reply api/proto/info/info.proto]: Agreed.

**Comment by eriknordmark** [comment]
> Good point.. Added it.

I don't see it. I assume you didn't push.

File: api/proto/info/info.proto (hunk 3 above)
Comment (kalyan-nidumolu) [reply api/proto/info/info.proto]: Looked at the code.. I see what. you are saying.. 

1) Change in App Instance - as long as the config is there - are sent in ZInfoApp messages
     - These are sent for every change in AppInstanceStatus

2) So - deviceInfo.AppInstance is mainly useful when the App Instance is deleted.
     - To see the various states the App is going through when it is deleted.
     - We are also sending the ZInfoApp for each of these triggers - So as long as cloud retains them, we don't need to look at the info in the device.

So based on this, I am thinking - lets add the run state, but trigger device info in AppInstanceStatus handler based on the following:

1) New AppInstanceStatus created --> Trigger devInfo
2) AppInstanceConfig deleted --> Trigger devInfo for all AIS changes
3) If AIC is present --> DO NOT trigger devInfo for AIS changes ( Except (1) above )

Does this sound good?

File: api/proto/info/info.proto (hunk 1 above)
Comment (kalyan-nidumolu) [reply api/proto/info/info.proto]: Changed it to 32.

**Comment by kalyan-nidumolu** [comment]
Took care of all the suggested changes.. This is now ready to merge..

**Pull request merged**